starter/src/main/java/io/javaoperatorsdk/operator/springboot/starter/CrdUploader.java

```diff
@@ -0,0 +1,91 @@
+package io.javaoperatorsdk.operator.springboot.starter;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.function.UnaryOperator;
+
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.client.KubernetesClient;
+
+import static java.util.Arrays.stream;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+
+public class CrdUploader {
+
+  private static final Logger log = LoggerFactory.getLogger(CrdUploader.class);
+
+  private static final int CRD_READY_WAIT = 2000;
+
+  private final CrdTransformer crdTransformer;
+  private final KubernetesClient kubernetesClient;
+  private final boolean applyOnStartup;
+  private final String crdSuffix;
+  private final String crdPath;
+
+  public CrdUploader(KubernetesClient kubernetesClient, List<CrdTransformer> transformers,
+      boolean applyOnStartup, String crdPath, String crdSuffix) {
+    this.crdTransformer = CrdTransformer.reduce(transformers);
+    this.kubernetesClient = kubernetesClient;
+    this.applyOnStartup = applyOnStartup;
+    this.crdSuffix = crdSuffix;
+    this.crdPath = crdPath;
+  }
+
+  public void upload() throws Exception {
+    if (applyOnStartup) {
```

Comment:
Would not be this possible just use `@ConditionalOnProperty`, and create a CrdUploader bean and autoamtically apply when apply on startup property is present.